util/src/main/java/io/grpc/util/AdvancedTlsX509TrustManager.java

```diff
@@ -303,27 +318,35 @@ public static Builder newBuilder() {
     return new Builder();
   }
 
-  // The verification mode when authenticating the peer certificate.
+  /**
+   * The verification mode when authenticating the peer certificate.
+   */
   public enum Verification {
-    // This is the DEFAULT and RECOMMENDED mode for most applications.
-    // Setting this on the client side will do the certificate and hostname verification, while
-    // setting this on the server side will only do the certificate verification.
+    /**
+     * This is the DEFAULT and RECOMMENDED mode for most applications.
+     * Setting this on the client side performs both certificate and hostname verification, while
+     * setting it on the server side only performs certificate verification.
+     */
     CERTIFICATE_AND_HOST_NAME_VERIFICATION,
-    // This SHOULD be chosen only when you know what the implication this will bring, and have a
-    // basic understanding about TLS.
-    // It SHOULD be accompanied with proper additional peer identity checks set through
-    // {@code PeerVerifier}(nit: why this @code not working?). Failing to do so will leave
-    // applications to MITM attack.
-    // Also note that this will only take effect if the underlying SDK implementation invokes
-    // checkClientTrusted/checkServerTrusted with the {@code SSLEngine} parameter while doing
-    // verification.
-    // Setting this on either side will only do the certificate verification.
+    /**
+     * This SHOULD be accompanied by proper additional peer identity checks set through
```

Comment:
The docs here and below say "how to use it" sorta but not much of "what it does". How do you feel about something like:

> DANGEROUS: Use trusted credentials to verify the certificate, but clients will not verify the certificate is for the expected host. This setting is only appropriate when accompanied by proper additional peer identity checks set through SslSocketAndEnginePeerVerifier. Failing to do so will leave your applications vulnerable to MITM attacks.
>
> <p>This setting has the same behavior on server-side as CERTIFICATE_AND_HOST_NAME_VERIFICATION. It is also the same behavior client-side when the caller does not use the X509ExtendedTrustManager APIs to provide the `SSLEngine` or `Socket`.

Reply:
I like all the first paragraph, but the second looks confusing - we describe the behavior by referencing to this class, and then to the base class. I think 'Setting this on either side will only perform certificate verification.' is more concise.
Also, I'd still like to add as the last statement the existing note:
>Keep in mind that this will only take effect if the underlying SDK implementation invokes checkClientTrusted/checkServerTrusted with the {@code SSLEngine} parameter while doing verification.

WDYT?